packages/apps/reputation-oracle/server/src/modules/abuse/abuse.service.spec.ts

```diff
@@ -0,0 +1,495 @@
+    abuseService = moduleRef.get<AbuseService>(AbuseService);
+    abuseRepository = moduleRef.get<AbuseRepository>(AbuseRepository);
+    webhookOutgoingService = moduleRef.get<WebhookOutgoingService>(
+      WebhookOutgoingService,
+    );
+    reputationService = moduleRef.get<ReputationService>(ReputationService);
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('reportAbuse', () => {
+    it('should create a new abuse entity', async () => {
+      const userId = 1;
+
+      await abuseService.reportAbuse({
+        escrowAddress,
+        chainId,
+        userId,
+      });
+
+      expect(abuseRepository.createUnique).toHaveBeenCalledWith({
+        escrowAddress: escrowAddress,
+        chainId: chainId,
+        userId: userId,
+        retriesCount: 0,
+        status: AbuseStatus.PENDING,
+        waitUntil: expect.any(Date),
+      });
+    });
+  });
+
+  describe('processSlackInteraction', () => {
+    it('should send an Abuse Report Modal to Slack if the decision is accepted', async () => {
+      const abuseEntity = generateAbuseEntity({ status: AbuseStatus.NOTIFIED });
+
+      const dto = {
+        callback_id: abuseEntity.id,
+        chainId,
+        type: 'interactive_message',
+        actions: [{ value: AbuseDecision.ACCEPTED }],
+        trigger_id: faker.string.uuid(),
+        response_url: faker.internet.url(),
+      };
+
+      jest
+        .spyOn(abuseRepository, 'findOneById')
+        .mockResolvedValueOnce(abuseEntity);
+
+      EscrowUtils.getEscrow = jest.fn().mockResolvedValueOnce({
+        launcher: fakeAddress,
+      });
+      const amount = faker.number.int();
+      OperatorUtils.getOperator = jest.fn().mockResolvedValueOnce({
+        amountStaked: amount,
+      });
+
+      await abuseService.processSlackInteraction(dto as any);
+
+      expect(abuseSlackBot.triggerAbuseReportModal).toHaveBeenCalledWith({
+        abuseId: abuseEntity.id,
+        escrowAddress: abuseEntity.escrowAddress,
+        chainId: abuseEntity.chainId,
+        maxAmount: amount,
+        triggerId: dto.trigger_id,
+        responseUrl: dto.response_url,
+      });
+    });
+
+    it('should update Slack message and entity if the decision is submitted via view_submission', async () => {
+      const abuseEntity = generateAbuseEntity({ status: AbuseStatus.NOTIFIED });
+
+      const dto = {
+        chainId,
+        type: 'view_submission',
+        view: {
+          callback_id: abuseEntity.id,
+          private_metadata: JSON.stringify({
+            responseUrl: faker.internet.url(),
+          }),
+          state: {
+            values: {
+              quantity_input: { quantity: { value: 10 } },
+            },
+          },
+        },
+      };
+
+      jest
+        .spyOn(abuseRepository, 'findOneById')
+        .mockResolvedValueOnce(abuseEntity);
+
+      await abuseService.processSlackInteraction(dto as any);
+
+      expect(abuseSlackBot.updateMessage).toHaveBeenCalledWith(
```

Comment:
Here and in other cases, would be also nice to check that some method is called exactly once